aspnet/mvc/controllers/filters.rst

```diff
@@ -1,8 +1,312 @@
+  :language: c#
+  :emphasize-lines: 12-15
+
+When this filter is applied to an action that also would have an action filter applied (later in the pipeline), such as the ``SomeResource`` action below, the action filter is not executed (in this case the ``AddHeader`` action filter is applied at the class level, but the behavior would be the same if it were applied to the method).
+
+.. literalinclude:: filters/sample/src/FiltersSample/Controllers/SampleController.cs
+  :language: c#
+  :emphasize-lines: 1,4
+  :lines: 6-8, 14-19
+  :dedent: 4 
+
+Configuring Filters
+-------------------
+Filters can be *scoped* at three different levels. You can add a particular filter to a particular action as an attribute. You can add a filter to all actions within a controller by applying an attribute at the controller level. Or you can register a filter globally, to be run with every MVC action.
+
+Filters are added globally in ``Startup``, when configuring MVC:
+
+.. literalinclude:: filters/sample/src/FiltersSample/Startup.cs
+  :language: c#
+  :emphasize-lines: 3-7
+  :lines: 15-24
+  :dedent: 8
+
+Filters can be added by type, or an instance can be added. If you add an instance, that instance will be used for every request. If you add a type, the instance will be created through DI, and any constructor dependencies will be populated by DI. In the example above, the :ref:`DurationActionFilter <duration-action-filter>` has a dependency on ``ILoggerFactory`` in its constructor, which is fulfilled by DI on each request.
+
+If you have a filter you don't want to be scoped globally, which requires dependencies to be provided through DI, you can apply it at the constructor or action level using the ``ServiceFilter`` or ``TypeFilter`` attribute. ``ServiceFilter`` lets you specify, as an attribute, the type of a filter you want to apply. It will instantiate the filter and any dependencies it has using DI, as shown in this example:
+
+.. literalinclude:: filters/sample/src/FiltersSample/Controllers/HomeController.cs
+  :language: c#
+  :emphasize-lines: 1
+  :lines: 19-24
+  :dedent: 8
+
+If you try to use ``ServiceFilter`` without registering the filter type in ``ConfigureServices``, you will get an exception like this one: *System.InvalidOperationException: No service for type 'FiltersSample.Filters.LoggingAddHeaderFilter' has been registered.* You can see that this type was registered in code example above - this is the important line:
+
+.. literalinclude:: filters/sample/src/FiltersSample/Startup.cs
+  :language: c#
+  :emphasize-lines: 1
+  :lines: 21
+  :dedent: 12
+
+``ServiceFilterAttribute`` implements `IFilterFactory <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Filters/IFilterFactory/index.html>`_, which exposes a single method for creating an ``IFilter`` instance. In the case of ServiceFilter, this is implemented by using the built-in service provider to load the type from DI.
```

Comment:
>In the case of ServiceFilter, this is implemented 

What is this? 

>by using the built-in service provider to load the type from DI.

Is it something like 
The create method (for creating an IFilter) is implemented by using the built-in service provider which loads the type from DI?  

I'm not understanding that sentence.